pkg/urlutil/urlutil_test.go

```diff
@@ -6,17 +6,21 @@ var (
 	gitUrls = []string{
 		"git://github.com/docker/docker",
 		"git@github.com:docker/docker.git",
-		"git@bitbucket.org:atlassianlabs/atlassian-docker.git",
+		"git@bitbucket.org:atlassianlabs/atlassian-docker",
 		"https://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git#branch",
 		"http://github.com/docker/docker.git#:dir",
+		"git+ssh://github.com/docker/docker",
+		"git+https://github.com/docker/docker.git#branch",
+		"git+ftps://github.com/docker/docker.git#:dir",
```

Comment:
It's not clear to me that `git+mail` should actually be treated differently than some other invalid URL (such as `git+https://github.com/.git`). The `git+` form always indicates a git URL, and all of these URIs are passed directly to git. I think it's fine to accept anything that git does.